zapisy/apps/enrollment/timetable/assets/components/PrototypeSummary.vue

```diff
@@ -0,0 +1,218 @@
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+import { values, cloneDeep } from "lodash";
+
+import state from "../store/groups";
+import { Course, Group } from "../models";
+import SimpleSummary from "./SimpleSummary.vue";
+
+@Component({
+  components: {
+    SimpleSummary,
+  },
+})
+export default class PrototypeSummary extends Vue {
+  public points: number = 0;
+  public howManyPointsCategories: number = 0;
+
+  public selectedGroups: Function = (group: Group, course: Course) =>
+    !group.isEnrolled &&
+    !group.isEnqueued &&
+    !group.isPinned &&
+    group.isSelected &&
+    group.course.id == course.id;
+
+  public pinnedGroups: Function = (group: Group, course: Course) =>
+    !group.isEnrolled &&
+    !group.isEnqueued &&
+    group.isPinned &&
+    group.course.id == course.id;
+
+  public enqueuedGroups: Function = (group: Group, course: Course) =>
+    group.isEnqueued && group.course.id == course.id;
+
+  public enrolledGroups: Function = (group: Group, course: Course) =>
+    group.isEnrolled && group.course.id == course.id;
+
+  public initializeCourses() {
+    let groups: { [id: number]: Group } = values(state.state.store).filter(
+      (g) => g.isEnrolled || g.isEnqueued || g.isPinned || g.isSelected
+    );
+    let enrolledCourses: { [id: number]: Course } = {};
+    let enrolledPoints: number = 0;
+    Object.entries(groups).forEach((g) => {
+      if (g[1].isEnrolled && enrolledCourses[g[1].course.id] === undefined) {
+        enrolledCourses[g[1].course.id] = cloneDeep(g[1].course);
+        enrolledCourses[g[1].course.id].summaryPoints =
+          enrolledCourses[g[1].course.id].points;
+        enrolledPoints += enrolledCourses[g[1].course.id].summaryPoints;
+      }
+    });
+    let enqueuedCourses: { [id: number]: Course } = {};
+    let enqueuedPoints: number = 0;
+    Object.entries(groups).forEach((g) => {
+      if (g[1].isEnqueued && enqueuedCourses[g[1].course.id] === undefined) {
+        enqueuedCourses[g[1].course.id] = cloneDeep(g[1].course);
+        if (enrolledCourses[g[1].course.id] === undefined) {
+          enqueuedCourses[g[1].course.id].summaryPoints =
+            enqueuedCourses[g[1].course.id].points;
+        } else {
+          enqueuedCourses[g[1].course.id].summaryPoints = -1;
+          enrolledPoints += 1;
+        }
+        enrolledPoints += enqueuedCourses[g[1].course.id].summaryPoints;
```

Comment:
O rany. Serio?
```suggestion
        if (enrolledCourses[g[1].course.id] === undefined) {
          enqueuedCourses[g[1].course.id].summaryPoints =
            enqueuedCourses[g[1].course.id].points;
          enrolledPoints += enqueuedCourses[g[1].course.id].summaryPoints;
        } else {
          enqueuedCourses[g[1].course.id].summaryPoints = -1;
        }
```
I tak samo w jeszcze dwóch analogicznych miejscach.